engine/core/src/main/java/org/datacleaner/job/builder/TransformerComponentBuilder.java

```diff
@@ -334,7 +334,9 @@ protected void onRemovedInternal() {
      *            a new change listener
      */
     public void addChangeListener(TransformerChangeListener listener) {
-        _localChangeListeners.add(listener);
+        if (!_localChangeListeners.contains(listener)) {
```

Comment:
This scares me. I think we ping-ponged on `localChangeListeners.contains` in various places quite a few times. Maybe check if this was one of those places, and if it was, why it was removed. I'm afraid of re-opening new bugs.